File: app/src/main/java/org/michaelbel/tjgram/data/entity/Medium.java
package org.michaelbel.tjgram.data.entity;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class Medium implements Serializable {
    public static final int TYPE_IMAGE = 1;
    public static final int TYPE_VIDEO = 2;

    @Expose @SerializedName("additionalData") public AdditionalData additionalData;
    @Expose @SerializedName("iframeUrl") public String iframeUrl;
    @Expose @SerializedName("imageUrl") public String imageUrl;
    @Expose @SerializedName("service") public String service;
    @Expose @SerializedName("size") public Size size;
    @Expose @SerializedName("type") public Integer type;
}

File: app/src/main/java/org/michaelbel/tjgram/data/entities/Medium.kt
package org.michaelbel.tjgram.data.entities

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class Medium (
    @Expose @SerializedName("additionalData") var additionalData: AdditionalData,
    @Expose @SerializedName("iframeUrl") var iframeUrl: String,
    @Expose @SerializedName("imageUrl") var imageUrl: String,
    @Expose @SerializedName("service") var service: String,
    @Expose @SerializedName("size") var size: Size,
    @Expose @SerializedName("type") var type: Int
): Serializable